### PR TITLE
[PD-2387] Add refresh indicator for reports

### PR DESCRIPTION
**Pull request by koonsies**
## Overview

This PR simply adds a refresh spinner to the ReportList when a report is refreshed

## Testing

* Do the dev-server/gulp shuffle.
* Head to dynamic reporting
* Click an existing report in the sidebar
* Click "refresh"

_NOTE: Locally, refresh happens very quickly for me. Adding a sleep to myreports->views->refresh_report helps to see the spinner_

### CODE REVIEW

File: gulp/src/common/ui/PopMenu.jsx [hunk 1]
```diff
@@ -5,19 +5,18 @@ export default class PopMenu extends Component {
   handleItemClick(item, e) {
     e.preventDefault();
     item.onSelect();
-    this.setState({isMenuActive: false});
```
Comment (koonsies) [inline gulp/src/common/ui/PopMenu.jsx]: I never saw that this was used.

File: gulp/src/common/ui/PopMenu.jsx [hunk 2]
```diff
@@ -5,19 +5,18 @@ export default class PopMenu extends Component {
...
+    const {options, isMenuActive, closeAllPopups, isMenuPending} = this.props;
     const dropdownItems = options.map((item, index)=> {
       return (
         <li key={index} onClick={e => this.handleItemClick(item, e)}>
           {item.display}
         </li>
       );
     });
-    if (isMenuActive === true) {
+    if (isMenuActive === true & isMenuPending === false) {
```
Comment (koonsies) [inline gulp/src/common/ui/PopMenu.jsx]: Is "===" the accepted method for existence checks/boolean comparisons? Is "===" preferable to "isMenuActive & !isMenuPending"?

File: gulp/src/common/ui/PopMenu.jsx [hunk 3]
```diff
@@ -31,7 +30,16 @@ export default class PopMenu extends Component {
...
+    const {toggleMenu, isMenuPending} = this.props;
+    if (isMenuPending){
+      return (
+        <div style={{position: 'relative'}}>
+          <span className="report-loader list-loader">
+          </span>
+          {this.menuContents()}
+        </div>
+      );
+    }
```
Comment (koonsies) [inline gulp/src/common/ui/PopMenu.jsx]: Is there a better way to handle this?

File: gulp/src/common/ui/PopMenu.jsx (hunk 2 above)
Comment (aspidites) [inline gulp/src/common/ui/PopMenu.jsx]: I'd prefer ! myself. That said, you should be using `&&` as  `&` is bitwise.

File: gulp/src/common/ui/PopMenu.jsx (hunk 3 above)
Comment (darrint) [inline gulp/src/common/ui/PopMenu.jsx]: You can call functions from inside the jsx:

      return (
        <div style={{position: 'relative'}}>
          {this.renderMenuStuff()}
          {this.menuContents()}
        </div>

    ...

    renderMenuStuff() {
      const {isMenuPending} = this.props;
      if (isMenuPending) {
        return <span pendingway .../>
      }
      return <span otherway .../>

**Pull request merged**